From 8127a1d15b63b1f582be9c212472b8d50bcb3b9d Mon Sep 17 00:00:00 2001
From: Sarthak Garg <quic_sartgarg@quicinc.com>
Date: Thu, 7 Nov 2024 13:35:03 +0530
Subject: [PATCH 06/18] dt-bindings: mmc: qcom: Document level shifter flag for
 SD card

Introduce a flag to indicate if the Qualcomm platform has a level
shifter for SD cards. With level shifter addition some extra delay is
seen on RX data path leading to CRC errors. To compensate these delays
and avoid CRC errors below things needs to be done:

1) Enable tuning for SDR50 mode
2) Limit HS mode frequency to 37.5MHz from 50MHz

Add this flag for all targets with a level shifter to handle these
issues for SD card.

Signed-off-by: Sarthak Garg <quic_sartgarg@quicinc.com>
---
 Documentation/devicetree/bindings/mmc/sdhci-msm.yaml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Documentation/devicetree/bindings/mmc/sdhci-msm.yaml b/Documentation/devicetree/bindings/mmc/sdhci-msm.yaml
index 594bd174f..14b91538a 100644
--- a/Documentation/devicetree/bindings/mmc/sdhci-msm.yaml
+++ b/Documentation/devicetree/bindings/mmc/sdhci-msm.yaml
@@ -138,6 +138,9 @@ properties:
     $ref: /schemas/types.yaml#/definitions/uint32
     description: platform specific settings for DLL_CONFIG reg.
 
+  qcom,use-level-shifter:
+    description: Flag to indicate if platform has level shifter for SD card.
+
   iommus:
     minItems: 1
     maxItems: 8
-- 
2.43.0

